cm_service_restart.py

```diff
@@ -56,11 +69,32 @@ def select_from_items(item_type, items):
             service_name = args.service.lower()
         else:
             service_name = select_from_items('service', service_names)
+
+        # Get the action to be performed on predefined actions
+        defined_actions = ['Service restart', 'Deploy Client config', 'Rolling Restart']
+
         for service in services:
             if service.name.lower() == service_name:
 
-                # Restart the service
-                print "Attempting to restart '%s' in '%s'" % (service_name, cluster_name)
-                command = service.restart().wait()
-                print command
-                sys.exit()
+                action = select_from_items('action', [i.lower() for i in defined_actions])
+
+                if action == defined_actions[0].lower():  # Restart the service
+                    command = service.restart().wait()
+
+                elif action == defined_actions[1].lower():    # Deploy client configs
+                    command = service.deploy_client_config().wait()
+
+                elif action == defined_actions[2].lower() and ('hue' in service.name.lower() or 'hive' in service.name.lower() or \
+                                                               'oozie' in service.name.lower() or 'spark' in service.name.lower()):    # RR unavailable for these
+                    print "Rolling Restart is unavailable in " + service.name.upper()
```

Comment:
I think i would rather just call it and then catch the exception - i think that's more pythonic.